pymc/aesaraf.py

```diff
@@ -535,55 +535,156 @@ def make_shared_replacements(point, vars, model):
 
 def join_nonshared_inputs(
     point: Dict[str, np.ndarray],
-    xs: List[TensorVariable],
-    vars: List[TensorVariable],
-    shared,
-    make_shared: bool = False,
-):
+    outputs: List[TensorVariable],
+    inputs: List[TensorVariable],
+    shared_inputs: Union[Dict[TensorVariable, TensorSharedVariable], None] = None,
+    make_inputs_shared: bool = False,
+) -> Tuple[List[TensorVariable], TensorVariable]:
     """
-    Takes a list of Aesara Variables and joins their non shared inputs into a single input.
+    Create new outputs and input TensorVariables where the non-shared inputs are joined
+    in a single raveled vector input.
 
     Parameters
     ----------
-    point: a sample point
-    xs: list of Aesara tensors
-    vars: list of variables to join
+    point: dictionary of variable names to numerical values
```

Comment:
The parameters should be documented with the following pattern:

```
parameter_name : parameter_type
    parameter description
```

Therefore, this would be:

```suggestion
    point : dict of {str : array_like}
```

It is then the description that explains the meaning of the strings and arrays in the parameter or if only a subset of objects with string and array type are valid. The description already does that so no changes are needed there. This page is a useful reference on docstrings for pymc: https://pymc-data-umbrella.xyz/en/latest/sprint/tutorials/docstring_tutorial.html#edit-the-docstring. The colon separating the name and the type of the parameter also needs to have one space both before and after for correct rendering.

Reply:
Thanks for this resource. I will take a look and make changes accordingly

Reply:
Reading over the link again. Should it just be `point : Point`?

Reply:
Also, I don't follow why the [Uniform example](https://www.pymc.io/projects/docs/en/latest/api/distributions/generated/pymc.Uniform.html#pymc.Uniform) doesn't have that format with the spacing. It is noted but isn't clear to me